### PR TITLE
AWS WAF stacks

### PR DESCRIPTION
**Pull request by txtchirag**

### CODE REVIEW

File: services/WAF/test_script/rate-attacks.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+token="eyJraWQiOiJsRVE1N29YdW9EYlhcL0ExK2tRa3RydEtQNzVrenJPeFZEK1liXC9ZZTNtUEk9IiwiYWxnIjoiUlMyNTYifQ.eyJzdWIiOiI0OWM4ZmFmYy1mY2Q0LTQxZjEtODE3Zi1kNTc1OWUwMzg0ZjciLCJhdWQiOiIyNmc1NWczaWtkbHRkYmJhMzhicTk1Z2UxcyIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJldmVudF9pZCI6IjMxOTllOTk5LTdkMjktNGM5NS05N2E2LTZiMDA1MmE1MjJhMCIsInRva2VuX3VzZSI6ImlkIiwiYXV0aF90aW1lIjoxNjAyNTY3NjQzLCJpc3MiOiJodHRwczpcL1wvY29nbml0by1pZHAudXMtZWFzdC0xLmFtYXpvbmF3cy5jb21cL3VzLWVhc3QtMV9MWHhTd1EwdUIiLCJjb2duaXRvOnVzZXJuYW1lIjoia2l0Y2h1NjY2QGdtYWlsLmNvbSIsImV4cCI6MTYwMjU3MTI0MywiaWF0IjoxNjAyNTY3NjQzLCJlbWFpbCI6ImtpdGNodTY2NkBnbWFpbC5jb20ifQ.SB3NlwsCZXO0Bnl6zCAFWUi38FLbYXzxQ8u3qncTZS-vB22Y-6WdsPEFGY8dHNREqGByJq9xbO_g0gfxC-Rs_tkpN-Nhf2DBAxHPwtQhbowDQq4RFZ4sCO3Am-Pva6XXl_r5_4UWRobIfsR3ihmyB-RuTVpnf1o7yuzIRNptmqIt5iluAfmbuocsNoTNrWvkT4su3IQiHIyZQIz3TzX6dIOGzuLD-sD9tjOAXwwaJSCjmZn4fAhe7bko-gOfTcalWkgPTRt7WEl4wlRl-rsrIWtriC8Q3ssw8-VnOP_T3b1XvH2jHS5DsG3TrKkgKdYybw6MfDq72kj0R8yytgOsuw"
```
Comment (jtdv01) [inline services/WAF/test_script/rate-attacks.sh:2]: Let's not commit any tokens in the repo. I'd suggest to have a placeholder instead.

```suggestion
token="${TOKEN}"
```

File: services/WAF/test_script/attack-result.log [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
..."Update": "Success"}{"Update": "Success"}{"Update": "Success"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}{"message":"Forbidden"}curl: (6) Could not resolve host: 6jztqidvd6.execute-api.us-east-1.amazonaws.com
```
Comment (jtdv01) [inline services/WAF/test_script/attack-result.log:2]: Nice! 😄

File: services/WAF/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+!!!!Note: need to test the below!!!!
+```
+# Build
+sam build
+
+# Try to invoke it
+sam local invoke
+
+# Deploy
+sam deploy
```
Comment (jtdv01) [inline services/WAF/README.md:22]: Very nice! Would be keen to see a demo of it 👍

**Review by jtdv01 — CHANGES_REQUESTED** [review]
Looking good! I'm just requesting a small change to remove the token, let's not commit any secrets in the repo.

File: services/WAF/test_script/rate-attacks.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+for (( ; ; )); do \
+curl 'https://6jztqidvd6.execute-api.us-east-1.amazonaws.com/qa/mysfits/33e1fbd4-2fd8-45fb-a42f-f92551694506/like' \
```
Comment (jtdv01) [inline services/WAF/test_script/rate-attacks.sh:4]: This can have a placeholder too

Comment (txtchirag) [reply services/WAF/test_script/rate-attacks.sh]: updated ....will call the script with  json/params

**Review by jtdv01 — APPROVED** [review]
Looks good to me.

The WAF could be deployed as a standard Cloudformation template - there is no need to use SAM.

You can follow this up by writing a wrapper script as part of `./bin/soar-stack.sh`

For example, someone can run this to build a WAF on their account:

```
./bin/soar-stack.sh create-waf-stack
```

**Pull request merged**